.github/workflows/build-and-test.yml

```diff
@@ -28,22 +22,30 @@ jobs:
       - name: Check out code into the Go module directory
         uses: actions/checkout@v2
 
-      - name: Build all on ${{matrix.os}}
+      - name: Build all on ubuntu-20.04
         run: make all
-        
+
+
+      - name: Install Snyk CLI
+        run: |
+          npm install -g snyk@1.1288.0
+
+      - name: Run Snyk security scan
+        run: |
+          snyk auth ${{ secrets.SNYK_TOKEN }}
+          snyk test --all-projects
+
+      - name: Run Snyk Monitor
+        run: |
+          snapshot_link=$(snyk monitor --all-projects | grep "Explore this snapshot at" | sed 's/Explore this snapshot at //')
+          echo "[Snyk Monitor link]($snapshot_link)" >> $GITHUB_STEP_SUMMARY
+
       - name: Upload test report
```

Comment:
now that we have merged the `publish` step, do we still need to uplaod and download the test report?